crates/core/tedge_mapper/src/tests.rs

```diff
@@ -168,6 +169,50 @@ async fn c8y_mapper_syncs_pending_alarms_on_startup() {
     c8y_mapper.abort();
 }
 
+#[tokio::test]
+#[serial]
+async fn c8y_mapper_event_mapping_to_smartrest() {
+    let broker = mqtt_tests::test_mqtt_broker();
+
+    let mut messages = broker.messages_published_on("c8y/s/us").await;
+
+    // Start the C8Y Mapper
+    let c8y_mapper = start_c8y_mapper(broker.port).await.unwrap();
+
+    let _ = broker
+        .publish_with_opts(
+            "tedge/events/click_event",
+            r#"{ "message": "Detected one click" }"#,
+            mqtt_channel::QoS::AtLeastOnce,
+            false,
+        )
+        .await
+        .unwrap();
+
+    let mut msg = messages
+        .recv()
+        .with_timeout(MESSAGE_TIMEOUT_MS)
+        .await
+        .expect_or("No message received before timeout");
+    dbg!(&msg);
+
+    // The first message could be SmartREST 114 for supported operations
+    if msg.contains("114") {
+        // Fetch the next message which should be the event message
+        msg = messages
+            .recv()
+            .with_timeout(MESSAGE_TIMEOUT_MS)
+            .await
+            .expect_or("No message received before timeout");
+    }
```

Comment:
I found this part of the code a little bit weird, I had encountered it during the #789 and maybe we could implement it as a check in the `test_broker`?

Reply:
Looking forward to see how you handle it in #789. I'll replicate the same here.